src/test/ui/intrinsics/is_const_eval.rs

```diff
@@ -0,0 +1,11 @@
+// run-pass
+
+#![feature(core_intrinsics)]
+use std::intrinsics::is_const_eval;
+
+fn main() {
+    const X: bool = unsafe { is_const_eval() };
+    let y = unsafe { is_const_eval() };
+    assert_eq!(X, true);
+    assert_eq!(y, false);
```

Comment:
Can you come up with an example where the intrinsic would be used as you imagine? I still believe you can't write anything new with this intrinsic (other than maybe the `eq` impl for slices)

Reply:
Done (see the other test).